test/App.spec.js

```diff
@@ -0,0 +1,46 @@
+import { mount } from '@vue/test-utils';
+import app from '../src/App.vue';
+
+describe('App.vue', () => {
+
+  const wrapper = mount(app);
+  it('renders the todo app', () =>{
+    expect(wrapper.contains('div')).toBe(true);
+  });
+
+  //for some unforsaken reason, this one refuses to work.
+  describe('when the toggleEditModeb function is called', () => {
+    const editButton = wrapper.findAll('button').at(2);
+    editButton.trigger('click');
```

Comment:
Always put that into a `beforeEach` or put that into your test case `it`.

When node reads this file, it will execute that code immediately. That's probably not what you want. You want to execute it before you run the test case below.

Do you know `beforeEach`? https://jestjs.io/docs/en/setup-teardown

Reply:
we know about setup and teardown. we put that area of code into the 'it' part because we only need this setup for one test

Reply:
Exactly, your code is in the `describe` block but not in the `it` block. Put the code in the `it` block. 

If you add just one more test case here, you will end up **with the mutated wrapper** from the test case before. But you want a fresh wrapper object. So re-initialize the `wrapper` in a `beforeEach` or put the code into the `it` block so it does no interfere with other test cases.